woocommerce/woocommerce
# [Accessibility] Make the rating stars accessible

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #51918 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to a product page.
2. Scroll down and open the reviews tab.
3. The rating stars behave now as a radio group, this means that:
    - If no star is selected, the focus ring should go to the first star.
    - Pressing tab if any star is focused should move the focus away from the stars group.
    - Once a star is focused, use the arrows on the keyboard to navigate through the stars and select them.
    - Screen readers should announce the focused star.
4. Verify the review functionality is working as before.
5. Verify the visual appearance of the rating stars didn't change per theme.
6. Test it across all Twenty* WP themes and Storefront.
7. **HEADS-UP**: The Storefront theme doesn't load Woo core stylesheet, so there is a visual regression that will need to be fixed on the Storefront code base.

**Before**:
![Screenshot 2024-11-22 at 17 22 48](https://github.com/user-attachments/assets/ec4a3f58-e858-4dc1-937b-f518eefcbe96)

**After**
![Screenshot 2024-11-22 at 17 23 46](https://github.com/user-attachments/assets/fd6a545c-af8b-4233-bb79-cec166476523)





### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
